Build error becauseof old_shift_pressed

## Version of OpenTTD
jgrpp-0.40.3

## Expected result
Build success

## Actual result
```
(...)
[ 71%] Building CXX object CMakeFiles/openttd.dir/src/video/allegro_v.cpp.o
/home/ottd/jgrpp/repo/src/video/allegro_v.cpp: In member function ‘virtual void VideoDriver_Allegro::InputLoop()’:
/home/ottd/jgrpp/repo/src/video/allegro_v.cpp:474:6: error: ‘old_shift_pressed’ was not declared in this scope
  if (old_shift_pressed != _shift_pressed) HandleShiftChanged();
      ^~~~~~~~~~~~~~~~~
/home/ottd/jgrpp/repo/src/video/allegro_v.cpp:474:6: note: suggested alternative: ‘_shift_pressed’
  if (old_shift_pressed != _shift_pressed) HandleShiftChanged();
      ^~~~~~~~~~~~~~~~~
      _shift_pressed
CMakeFiles/openttd.dir/build.make:2915: recipe for target 'CMakeFiles/openttd.dir/src/video/allegro_v.cpp.o' failed
make[2]: *** [CMakeFiles/openttd.dir/src/video/allegro_v.cpp.o] Error 1
CMakeFiles/Makefile2:1330: recipe for target 'CMakeFiles/openttd.dir/all' failed
make[1]: *** [CMakeFiles/openttd.dir/all] Error 2
Makefile:181: recipe for target 'all' failed
make: *** [all] Error 2
```

## Steps to reproduce
I'm not aware of code but I think it seems that ``old_shift_pressed`` is not defined.
https://github.com/JGRennison/OpenTTD-patches/blob/f81c71e15b04bcf6202d27242af9e3803b439191/src/video/allegro_v.cpp#L474

--- a/src/video/allegro_v.cpp
+++ b/src/video/allegro_v.cpp
@@ -451,6 +451,7 @@ void VideoDriver_Allegro::Stop()
 void VideoDriver_Allegro::InputLoop()
 {
 	bool old_ctrl_pressed = _ctrl_pressed;
+	bool old_shift_pressed = _shift_pressed;
 
 	_ctrl_pressed  = !!(key_shifts & KB_CTRL_FLAG) != _invert_ctrl;
 	_shift_pressed = !!(key_shifts & KB_SHIFT_FLAG) != _invert_shift;

